packages/ui-components/src/components/time-picker-calendar/time-picker-calendar.tsx

```diff
@@ -0,0 +1,302 @@
+		if (newDisabledDates !== undefined && !areDatesValid(newDisabledDates)) {
+			throw new Error('Disabled dates should be an array with valid dates');
+		}
+	}
+
+	componentWillLoad() {
+		this.validateSelectedDates(this.selectedDates);
+		this.validateInitialDate(this.initialDate);
+		this.validateDisabledDates(this.disabledDates);
+	}
+
+	public onClickPreviousMonth = (event: MouseEvent) => {
+		if (this.month === 1) {
+			this.year = this.year - 1;
+			this.month = 12;
+
+			this.changeMonth.emit({ event: event, payload: this.month });
+			this.changeYear.emit({ event: event, payload: this.year });
+
+			return;
+		}
+
+		this.month = this.month - 1;
+		this.changeMonth.emit({ event: event, payload: this.month });
+	};
+
+	public onClickNextMonth = (event: MouseEvent) => {
+		if (this.month === 12) {
+			this.year = this.year + 1;
+			this.month = 1;
+
+			this.changeMonth.emit({ event: event, payload: this.month });
+			this.changeYear.emit({ event: event, payload: this.year });
+
+			return;
+		}
+
+		this.month = this.month + 1;
+		this.changeMonth.emit({ event: event, payload: this.month });
+	};
+
+	public isDayDisabled = (day: number): boolean => {
+		const dayMoment = fromDateFields(day, this.month, this.year);
+
+		if (isDateInArray(dayMoment, this.disabledDates)) {
+			return true;
+		}
+
+		if (this.minDate && isDateBefore(dayMoment, this.minDate)) {
+			return true;
+		}
+
+		if (this.maxDate && isDateAfter(dayMoment, this.maxDate)) {
+			return true;
+		}
+
+		return false;
+	};
+
+	public onClickDay = (day: number, event: MouseEvent): void => {
+		const clickedDateMoment = fromDateFields(day, this.month, this.year);
+		this.clickDate.emit({ event, payload: formatDate(clickedDateMoment) });
+	};
+
+	public isDayActive = (day: number): boolean => {
+		const date = fromDateFields(day, this.month, this.year);
+
+		return isDateInArray(date, this.selectedDates);
+	};
+
+	public onMouseEnter = (day: number): void => {
+		this.hoveredDay = day;
+		const date = fromDateFields(day, this.month, this.year).format(DATE_FORMAT);
+		this.hoveredDateChange.emit(date);
+	};
+
+	public onMouseLeave = (_day: number): void => {
+		this.hoveredDay = undefined;
+		this.hoveredDateChange.emit('');
+	};
+
+	public getSelectedRange = (): SelectedRange => {
+		if (this.selectedDates && this.selectedDates.length > 0) {
+			const startDate = this.selectedDates[0];
+
+			if (this.selectedDates.length === 1) {
+				return [startDate];
+			}
+
+			const endDate = this.selectedDates[this.selectedDates.length - 1];
+
+			return [startDate, endDate];
+		}
+
+		return [];
+	};
+
+	public isDayInRange = (day: number): boolean => {
+		const [selectedStartDate, selectedEndDate] = this.getSelectedRange();
+
+		if (this.isDayDisabled(day) || selectedStartDate === undefined || selectedEndDate !== undefined) {
+			return false;
+		}
+
+		const date = fromDateFields(day, this.month, this.year).format(DATE_FORMAT);
+		const hoveredDate = this.getHooveredDate();
```

Comment:
why do we need to control the hover here? Probably we should have another property inside the component to hover the day in other way and don't reuse this function to use that.

Reply:
We need this here to update the days that should be with a different color meaning that they belong to a specific range.